src/Yardarm/Spec/Path/PathParser.cs

```diff
@@ -21,6 +21,11 @@ public static PathSegment[] Parse(string path)
 
         private static IEnumerable<PathSegment> ParseInternal(string path)
         {
+            if (path.StartsWith("/"))
```

Comment:
I think starting with index = 1 would be a bit more efficient than allocating a new string on the heap.